protostar/commands/test/test_results.py

```diff
@@ -78,6 +78,23 @@ def from_failed_test_case_result(
         )
 
 
+@dataclass(frozen=True)
+class SetupCaseResult(TestResult, TimedTestResult):
+    test_case_name: str
+    setup_case_name: str
+
+
+@dataclass(frozen=True)
+class PassedSetupCaseResult(SetupCaseResult):
+    pass
+
+
+# Note: We rely on the fact that this class inherits from FailedTestCaseResult in formatting code.
```

Comment:
This seems a bit weird to me. We provide a distiction and then we tie it up in some confusing way.
Can't we just add?
```
    if isinstance(test_result, FailedSetupCaseResult):
        return _format_failed_test_case_result(test_result)
```
Is there a reason which prevents us from doing that?

Reply:
Done